client/llb/exec.go

```diff
@@ -438,15 +438,19 @@ func (e *ExecOp) Output() Output {
 }
 
 func (e *ExecOp) Inputs() (inputs []Output) {
-	mm := map[Output]struct{}{}
+	// make sure mounts are sorted
+	sort.Slice(e.mounts, func(i, j int) bool {
```

Comment:
Does this need to be sorted here to make the outputs consistent?
Thinking if I define the exec op a certain way I'd expect the, in this case here, mounts to be in that order.

Or are the way the mounts are added non-deterministic?

Reply:
So, I investigated this. The unit test I was running was marshaling the same `ExecOp` object multiple times. During the marshaling process, `mounts` is mutated and sorted in place. Thus, on subsequent marshals of the same ExecOp (in memory), the traversal order of the op graph will be different since `e.mounts` will have been mutated on the previous run. This is fixed, though, if we always sort the mounts before doing the inputs traversal as is the case here